mm/2s2h/Enhancements/Trackers/ItemTracker/ItemTrackerSettings.cpp

```diff
@@ -0,0 +1,611 @@
+                                   .Format("Scale: %.1f")
+                                   .Step(0.5f)
+                                   .Color(WIDGET_COLOR)
+                                   .Size(ImVec2(ImGui::GetContentRegionAvail().x, 0)))) {
+        windowObject.windowScale = scale;
+    }
+    if (UIWidgets::SliderFloat("Opacity", &opacity,
+                               UIWidgets::FloatSliderOptions()
+                                   .Min(0)
+                                   .Max(1.0f)
+                                   .DefaultValue(0.5f)
+                                   .LabelPosition(UIWidgets::LabelPosition::None)
+                                   .Format("Opacity: %.1f")
+                                   .Step(0.1f)
+                                   .Color(WIDGET_COLOR)
+                                   .Size(ImVec2(ImGui::GetContentRegionAvail().x, 0)))) {
+        windowObject.windowOpacity = opacity;
+    }
+    ImGui::PopID();
+    UIWidgets::Separator();
+}
+
+void ApplyRandoItemGroup(std::string listName) {
+    std::tuple<int16_t, int16_t, int16_t> list = randoItemLists.at(listName);
+    TrackerItemListObject itemObject = {
+        .windowName = listName,
+        .columnLength = std::get<2>(list),
+        .windowScale = 1.0f,
+        .windowOpacity = 0.5f,
+    };
+    for (int i = std::get<0>(list); i <= std::get<1>(list); i++) {
+        itemObject.itemList.push_back(i);
+    }
+
+    BenGui::mItemTrackerWindow->randoItemWindows.push_back(itemObject);
+}
+
+void ApplyDefaultItemGroup(std::string listName) {
+    std::tuple<int16_t, int16_t, int16_t> list = defaultItemLists.at(listName);
+    TrackerItemListObject itemObject = {
+        .windowName = listName,
+        .columnLength = std::get<2>(list),
+        .windowScale = 1.0f,
+        .windowOpacity = 0.5f,
+    };
+
+    if (listName == "Bottles") {
+        for (int j = std::get<0>(list); j <= std::get<1>(list); j++) {
+            int16_t bottle = ITEM_BOTTLE_1 + (j - ITEM_BOTTLE_1);
+            itemObject.itemList.push_back(bottle);
+        }
+    } else if (listName == "Tokens") {
+        for (int j = std::get<0>(list); j <= std::get<1>(list); j++) {
+            int16_t skullToken = ITEM_SKULL_TOKEN_SWAMP + (j - ITEM_SKULL_TOKEN_SWAMP);
+            itemObject.itemList.push_back(skullToken);
+        }
+    } else if (listName == "Stray Fairies") {
+        for (int j = std::get<0>(list); j <= std::get<1>(list); j++) {
+            int16_t strayFairy = ITEM_WOODFALL_STRAY_FAIRY + (j - ITEM_WOODFALL_STRAY_FAIRY);
+            itemObject.itemList.push_back(strayFairy);
+        }
+    } else if (listName == "Dungeon") {
+        for (int j = std::get<0>(list); j <= std::get<1>(list); j++) {
+            int16_t dungeonItem = ITEM_WOODFALL_DUNGEON_MAP + (j - ITEM_WOODFALL_DUNGEON_MAP);
+            itemObject.itemList.push_back(dungeonItem);
+        }
```

Comment:
These are all almost exactly the same, aside from the min and max values, which are defined in the very list referenced here. This could be condensed down to:

```suggestion
    if (listName == "Bottles" || listName == "Tokens" || listName == "Stray Fairies" || listName == "Dungeon") {
        int16_t baseValue = std::get<0>(list);
        int16_t maxValue = std::get<1>(list);
        for (int j = baseValue; j <= maxValue; j++) {
            itemObject.itemList.push_back(j);
        }
```